packages/core/src/messagingProtocols/mls/EventHandler/events/messageAdd/messageAdd.ts

```diff
@@ -38,10 +38,17 @@ const handleMLSMessageAdd = async (
 ): EventHandlerResult => {
   const encryptedData = Decoder.fromBase64(event.data).asBytes;
 
-  const groupId = await mlsService.getGroupIdFromConversationId(
-    event.qualified_conversation ?? {id: event.conversation, domain: ''},
-    event.subconv,
-  );
+  const qualifiedConversationId = event.qualified_conversation ?? {id: event.conversation, domain: ''};
+
+  const groupId = await mlsService.getGroupIdFromConversationId(qualifiedConversationId, event.subconv);
+
+  // We should not receive a message for a group the client is not aware of
+  if (!groupId) {
+    throw new Error(
+      `Could not find a group_id for conversation ${qualifiedConversationId.id}@${qualifiedConversationId.domain}`,
+    );
+  }
+
```

Comment:
Since `getGroupIdFromConversationId` does not throw anymore, we have to throw an error here (check out the comment in the code above)